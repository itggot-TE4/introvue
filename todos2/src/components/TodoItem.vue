<template>
  <li>
    <input
      :checked="todoItem.isCompleted"
      @change="$emit('toggle-completed', todoItem.id)"
      type="checkbox"
    />
    <span @click="showTodo" :class="{ 'is-complete': todoItem.isCompleted }">{{
      todoItem.title
    }}</span>
    <button @click="delTodo">X</button>
    {{ todoItem.title }}
  </li>
</template>

<script>
export default {
  name: "TodoItem",
  props: ["todoItem"],
  methods: {
    showTodo() {
      this.$router.push({ name: "Todo", params: { id: this.todoItem.id } });
    },
    delTodo() {
      fetch(`http://localhost:9292/api/v1/todos/${this.todoItem.id}`, {
        method: "delete",
        headers: {
          "content-type": "application/json"
        }
      }).then(json => this.$store.commit('delTodo', this.todoItem.id));
    }
  }
};
</script>

<style>
.is-complete {
  text-decoration: line-through;
}
</style>
