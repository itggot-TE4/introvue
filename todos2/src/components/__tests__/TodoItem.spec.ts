import { shallowMount } from "@vue/test-utils";
import TodoItem from "@/components/TodoItem.vue";

describe("TodoItem.vue", () => {
  it("renders props.todoItem.title when passed", () => {
    const todoItem = {
      title: "hello, world!"
    }
    const wrapper = shallowMount(TodoItem, {
      props: { todoItem }
    });
    expect(wrapper.text()).toMatch(todoItem.title);
  });

  it("does not render props.todoItem.description when passed", () => {
    const todoItem = {
      description: "hello, world!"
    }
    const wrapper = shallowMount(TodoItem, {
      props: { todoItem }
    });
    expect(wrapper.text()).not.toMatch(todoItem.description);
  });

  it("navigates to route /todos/<id> when clicked on", () => {
    const todoItem = {
      id: 1,
      title: "hello, world!"
    }

    const pushFunction = jest.fn();

    const wrapper = shallowMount(TodoItem, {
      global: {
        mocks: {
          $router: {
            push: pushFunction
          }
        }
      },
      props: { todoItem }
    });
    wrapper.find('span').trigger("click");
    expect(pushFunction).toHaveBeenCalledWith(expect.objectContaining({ name: "Todo" }));
    expect(pushFunction).toHaveBeenCalledWith({ name: "Todo", params: { id: todoItem.id } });
  });

  it("renders correctly", () => {
    const todoItem = {
      id: 1,
      title: "hello, world!",
      description: "foobar",
      isCompleted: true
    }
    const wrapper = shallowMount(TodoItem, {
      props: { todoItem }
    });

    expect(wrapper.html()).toMatchSnapshot();
})
});
