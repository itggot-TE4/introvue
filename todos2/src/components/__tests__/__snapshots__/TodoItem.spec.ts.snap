// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem.vue renders correctly 1`] = `<li><input type="checkbox"><span class="is-complete">hello, world!</span><button>X</button> hello, world!</li>`;
